### DISCUSSION TITLE
Comments in strings / regexps

### DISCUSSION THREAD

**Discussion by m-u-xyz** [textX/Arpeggio#123] [Q&A]
I'm trying to write a parser (in "cleanpeg" syntax) that accepts comments.
```
BlockComment = r'/\\*.*?\\*/'
LineComment = r'//~[\r\n]*'
Comment = LineComment | BlockComment
```
This `.peg` file doesn't parse because, surprise, the `//` in the second line introduces a comment even when it's inside a string.

Is there a way to tell arpeggio not to check for comments in some contexts? While working around this snag is fairly easy, the user of my DSL is going to have the same problem and definitely won't expect (or in fact accept) this behavior.

**Comment by igordejanovic** [comment]
Fixed in #125. Thanks for reporting.
